### ISSUE TITLE
Unexpected error for `throw null!;`

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#31294]
**Version Used**: 

Branch C# 8.0: Nullable reference type (29 Oct 2018)
https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types.md
Latest commit 2f8fef by AlekseyTs:
Compare Nullable modifiers while comparing type symbols by default. (#30770)

**Steps to Reproduce**:

Compile the following code
```cs
class C
{
    void M1() => throw null;
    void M2() => throw null!;
}
```

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGhhCmRgB8ABAJgAIAwgFgAUAG8x/af24AWfgFkAjAAoAlPwC8APn4wAFggD2Ad35lsWANxSZ8pbw3a9hk+cs4AhLdEBfIA

**Expected Behavior**:

No errors or error with clear error message

**Actual Behavior**:

`error CS0155: The type caught or thrown must be derived from System.Exception`

**Comment by jaredpar** [comment]
Calling it out because it took me a second to see but it's the `!` on the second `throw null!` that causes the issue here.

**Comment by jcouv** [comment]
When we fix this, we should also test that `x == null !` does the right thing (`null` should be recognized, despite the `!`). There may be other scenarios where `null` (or `default`) is treated specially, which we should verify.

**Comment by jcouv** [comment]
FYI @jaredpar I self-assigned this bug and moved to preview2, since this family of problems can manifest as a crash.

**Comment by chsienki** [comment]
Another from the same family: #31296
